lib/rdoc/web_ri.rb

```diff
@@ -0,0 +1,170 @@
+# frozen_string_literal: true
+require 'open-uri'
+require 'nokogiri'
+include Nokogiri
+
+require_relative '../rdoc'
+
+# A class to display Ruby HTML documentation.
+class RDoc::WebRI
+
+  # Where the documentation lives.
+  ReleasesUrl = 'https://docs.ruby-lang.org/en/'
+
+
+  def initialize(target_name, options)
+    release_name = get_release_name(options[:release])
+    entries = get_entries(release_name)
```

Comment:
So we'd have sent 2 requests by this point just to get the release and entries list. Considering both of them are rarely changed, I think we may introduce something like [rdoc-data](https://github.com/ruby/rdoc-data) used to do that packages all these information into a gem. And a tool like this can simply use it to generate the target url without making additional requests.

Reply:
On my machine(!), making the two requests is much faster than reading the static RI files.

So a new gem would include the data that this app gathers from the two requests?  And thus not need nokogiri?  Interesting.  But would not have master (though I'm not sure that's a drawback).

Reply:
The more think about this idea (the gem, already containing the data for each release), the better I like it.  Will explore.